tests/check_for_valid_readme.py

```diff
@@ -0,0 +1,53 @@
+"""Test script to check README.rst files for syntax errors."""
+import argparse
+import sys
+from pathlib import Path
+
+import readme_renderer.rst
+
+
+def is_valid_rst(path):
+    """Checks if RST can be rendered on PyPI."""
+    with open(path) as f:
+        markup = f.read()
+    if readme_renderer.rst.render(markup) is None:
+        return False
+    return True
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Checks README.rst file in path for syntax errors."
+    )
+    parser.add_argument(
+        "paths", nargs="+", help="paths containing a README.rst to test"
+    )
+    parser.add_argument("-v", "--verbose", action="store_true")
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    args = parse_args()
+
+    error = False
+    all_readmes_found = True
+
+    for path in [Path(path) for path in args.paths]:
+        readme = path / "README.rst"
+        if not readme.exists():
```

Comment:
I would move this check under the `if not is_valid_rst` (or in a try/except around it). We should first try to parse the readme, and if parsing failed we check to see if a nonexistent file was the reason.